### PR TITLE
1785/2138: Fixing and Improving Round Report Player Stats

### LINKED ISSUE #1785: 0.46.1 - Player that joins a team midgame does not get added to the round reports
In the picture below, Soup is not listed in the BV because they joined in round 2. I believe that any unit that is given to another player should be removed from the original player's BV calculation and added to the new player's BV calculation, and the other players also though that was the case.
image

### LINKED ISSUE #2138: RFE:  Add total units and percentage of units remaining to /who and round summary
MekHQ ATB has multiple win conditions that require tracking various forces by units, rather than BV.  I currently manually track these tallies turn by turn, per force/team.  I would like to have this tracked automatically by Megamek during gameplay as part of the summary. 

Current:
Kendricke's Templars: 16400 BV remaining (100.0% of 16400 initially) 0 BV fled
17th Donegal Guards: 12859 BV remaining (99.78% of 12887 initially) 0 BV fled
Thirty Weight Crew: 11871 BV remaining (98.68% of 12030 initially) 0 BV fled

Requested:
Kendricke's Templars: 16400 BV remaining (100.0% of 16400 initially) 0 BV fled, 8/8 units remaining (100%)
17th Donegal Guards: 12859 BV remaining (99.78% of 12887 initially) 0 BV fled, 12/12 units remaining (100%)
Thirty Weight Crew: 11871 BV remaining (98.68% of 12030 initially) 0 BV fled, 12/12 units remaining (100%)

### PR DESCRIPTION
**Pull request by Windchild292**
This fixes #1785 and fixes #2138 by adding initial unit count tracking, fixing the observer check, and having both change when the one traitors a unit.

### CODE REVIEW

File: megamek/i18n/megamek/common/report-messages_ru.properties [hunk 1]
```diff
@@ -815,7 +815,6 @@
 7005=Победитель: Чикаго Кабс!!!<newline>
 7010=Победитель: <data><newline>
 7015=Победитель: КОМАНДА #<data><newline>
-7016=<data>: <data> BV осталось (из <data> изначальных) <data> BV убежало
```
Comment (NickAragua) [inline megamek/i18n/megamek/common/report-messages_ru.properties:818]: <data>: <data> BV осталось (из <data> изначальных) <data> BV убежало. <data> / <data> бойцов осталось (<data>%).

Comment (NickAragua) [reply megamek/i18n/megamek/common/report-messages_ru.properties:818]: Can't do a suggestion on deleted lines, but that's a translation.

**Review by NickAragua — CHANGES_REQUESTED** [review]
Looks good, got a russian translation for you.

File: megamek/i18n/megamek/common/report-messages_ru.properties (hunk 1 above)
Comment (Windchild292) [reply megamek/i18n/megamek/common/report-messages_ru.properties:818]: Done, thanks for the translation. Merging based on the overall reply.

**Pull request merged**